cmd/generate/runner.go

```diff
@@ -54,9 +56,20 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
 	{
 		var vaultClient *vaultapi.Client
 		{
-			vaultClient, err = createVaultClientUsingEnv(ctx)
-			if err != nil {
-				return microerror.Mask(err)
+			if r.flag.VaultSecretName != "" && r.flag.VaultSecretNamespace != "" {
+				vaultClient, err = createVaultClientUsingK8sSecret(ctx, r.flag.VaultSecretNamespace, r.flag.VaultSecretName)
+				if errors.Is(err, rest.ErrNotInCluster) {
+					// fallthrough
+				} else if err != nil {
+					return microerror.Mask(err)
+				}
```

Comment:
I'd say this should also be the error. What is the situation where you specify the flags and it doesn't run in cluster _and_ you want that?

```suggestion
				if err != nil {
					return microerror.Mask(err)
				}
```